### PR TITLE
Python3: fix use a string pattern on a bytes object

### PR DESCRIPTION
**Pull request by dzhengfy**
Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

**Comment by dzhengfy** [comment]
When we run libvirt migration case, we got an error:
(root-BuDEOXnJ) [root@dzheng-test-machine root-BuDEOXnJ]# bin/avocado run --vt-type libvirt virsh.migrate.there_p2p.without_cpu_hotplug.compat_migration.non_compat_migration.with_postcopy
Log:
_2018-05-15 15:17:10,620 stacktrace       L0044 ERROR| Reproduced traceback from: /home/dzheng/source/avocado/avocado/core/test.py:864
2018-05-15 15:17:10,648 stacktrace       L0047 ERROR| Traceback (most recent call last):
2018-05-15 15:17:10,648 stacktrace       L0047 ERROR|   File "/home/dzheng/source/avocado-vt/avocado_vt/test.py", line 269, in runTest
2018-05-15 15:17:10,648 stacktrace       L0047 ERROR|     raise self.__status  # pylint: disable=E0702
2018-05-15 15:17:10,648 stacktrace       L0047 ERROR|   File "/home/dzheng/source/avocado-vt/avocado_vt/test.py", line 246, in setUp
2018-05-15 15:17:10,648 stacktrace       L0047 ERROR|     self._runTest()
2018-05-15 15:17:10,649 stacktrace       L0047 ERROR|   File "/home/dzheng/source/avocado-vt/avocado_vt/test.py", line 422, in _runTest
2018-05-15 15:17:10,649 stacktrace       L0047 ERROR|     env_process.postprocess(self, params, env)
2018-05-15 15:17:10,649 stacktrace       L0047 ERROR|   File "/home/dzheng/source/avocado-vt/virttest/error_context.py", line 135, in new_fn
2018-05-15 15:17:10,649 stacktrace       L0047 ERROR|     return fn(*args, **kwargs)
2018-05-15 15:17:10,649 stacktrace       L0047 ERROR|   File "/home/dzheng/source/avocado-vt/virttest/env_process.py", line 1412, in postprocess
2018-05-15 15:17:10,649 stacktrace       L0047 ERROR|     raise RuntimeError("Failures occurred while postprocess:\n%s" % err)
2018-05-15 15:17:10,649 stacktrace       L0047 ERROR| RuntimeError: Failures occurred while postprocess:
2018-05-15 15:17:10,649 stacktrace       L0047 ERROR| 
2018-05-15 15:17:10,649 stacktrace       L0047 ERROR| nfs cleanup: can't use a string pattern on a bytes-like object
2018-05-15 15:17:10,649 stacktrace       L0048 ERROR|_

**Comment by dzhengfy** [comment]
Refer to another merged PR: https://github.com/avocado-framework/avocado/pull/2617, this fix used same solution.

**Comment by dzhengfy** [comment]
Please consider this one because all of our migration cases will fail without this fix.

File: avocado/utils/software_manager.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def _init_on_demand(self):
                                'dnf': DnfBackend,
                                'zypper': ZypperBackend}
 
-            if backend_type not in backend_mapping.keys():
+            if backend_type not in list(backend_mapping.keys()):
```
Comment (ldoktor) [inline avocado/utils/software_manager.py]: This shouldn't be needed in py2 nor py3. In what version this breaks?

Comment (ldoktor) [reply avocado/utils/software_manager.py]: yes, but only where list is necessary. The `dict_keys` object implements `__in__` operator. Also the `keys` is also not needed at all as on all pythons the `in` operator on `dict` implies `is in keys`. So I'd just leave it as is or, in a separate commit, remove the `keys` completely.

Comment (dzhengfy) [reply avocado/utils/software_manager.py]: @ldoktor , ok, thanks for your explanation. I have updated.

**Comment by ldoktor** [comment]
Yep, I just verified it and (at least on my machine) the encoding of all the commands is `UTF-8`, therefor the only missing fix here is to turn all the `results.stdout` to `result.stdout_text` calls. Would you please take care of that?

**Comment by clebergnu** [comment]
@dzhengfy given that there's been no update to this PR, I'm closing this one, and waiting for a new version with the changes requested. Thanks!

**Pull request closed without merging**